src/xds.rs

```diff
@@ -183,17 +183,7 @@ impl ProxyStateUpdateMutator {
         // remove workload by UID; if xds_name is a service then this will no-op
         if let Some(prev) = state.workloads.remove(&strng::new(xds_name)) {
             // Also remove service endpoints for the workload.
-            for wip in prev.workload_ips.iter() {
-                let prev_addr = &network_addr(prev.network.clone(), *wip);
-                state
-                    .services
-                    .remove_endpoint(&prev.uid, &endpoint_uid(&prev.uid, Some(prev_addr)));
-            }
-            if prev.workload_ips.is_empty() {
-                state
-                    .services
-                    .remove_endpoint(&prev.uid, &endpoint_uid(&prev.uid, None));
-            }
+            state.services.remove_endpoint(&prev.uid, &prev.uid);
```

Comment:
Sending the exact same parameter twice looks odd. Can we just clean this up to only require 1 parameter now?

Reply:
nice, thanks